### PR TITLE
Adds the emp bomb, and a beacon to summon it.

### PR DESCRIPTION
**Pull request by Qwertytoforty**
## What Does This PR Do

Adds the EMP bomb and bomb core, a syndicate bomb style bomb that currently has a 18 36 EMP range. It can be var edited to adjust the size of emp it makes, or the number of pulses it makes (defaults to 1) as emps are destructive on guns and ipcs, but multiple emps must be used to take out borgs or mechs. Also adds a beacon to summon it, and make the default bomb summon beacons to have a variable to what bomb they summon. This is available to traitors for 10 TC.

Fixes incorrect fact of syndicate bombs in uplink description, minimum timer is 90 seconds, not 60.

## Why It's Good For The Game

More fun emp tools are fun. Can already be replicated by a chemical bomb core, but this is easier. Sprites for it are nice as well. Requested by Kyet.

## Images of changes

image

## Changelog
:cl:
add: Adds the EMP bomb, a syndicate bomb style explosive with a large EMP range.
tweak: Makes the bomb summon beacons have a variable that can be edited for what they summon.
/:cl:

### CODE REVIEW

File: code/game/machinery/syndicatebomb.dm [hunk 1]
```diff
@@ -449,6 +455,27 @@
...
+		log_game(adminlog)
+		adminloged = TRUE
+	empulse(src, heavy_emp, light_emp, 1)
+	if(pulse_number <= 1)
+		if(loc && istype(loc, /obj/machinery/syndicatebomb))
+			qdel(loc)
+		qdel(src)
+	else
+		pulse_number -= 1
+		addtimer(CALLBACK(src, .proc/detonate), 20) // every 2 seconds go off again till pulses run out
```
Comment (S-CR) [inline code/game/machinery/syndicatebomb.dm:477]: Add some indication to the bombcore that it is burnt out, perhaps?

Comment (Qwertytoforty) [reply code/game/machinery/syndicatebomb.dm:477]: Sure, one moment

**Comment by TDSSS** [comment]
I think this would be better added to traitor/nukie uplinks, instead of just leaving it for admemery. It's going to be used for a bit and then forgotten and lost to code if we don't.

**Comment by Qwertytoforty** [comment]
Any idea on a price then?

**Comment by FreeStylaLT** [comment]
inb4 IPC mains

**Review by Kyep — APPROVED** [review]

File: code/game/machinery/syndicatebomb.dm [hunk 2]
```diff
@@ -449,6 +455,33 @@
...
 	range_flame = 2
 
+/obj/item/bombcore/emp
+	name = "EMP bomb core"
+	var/light_emp = 36
+	var/heavy_emp = 18
+	var/pulse_number = 1 //Since one EMP wont destroy anything other then consoles and IPCS, here is an option to have multiple pulses when dentonating. DO NOT USE THIS WITH REALLY LARGE AREAS
+	var/adminloged = FALSE //If it exploded once, don't do it again.
+
+/obj/item/bombcore/emp/ex_act(severity) //It's an EMP bomb, not an chemical explosive
```
Comment (SteelSlayer) [inline code/game/machinery/syndicatebomb.dm:465]:
```suggestion
/obj/item/bombcore/emp/ex_act(severity) //It's an EMP bomb, not a chemical explosive
```

File: code/game/machinery/syndicatebomb.dm [hunk 3]
```diff
@@ -449,6 +455,33 @@
 	range_light = 4
 	range_flame = 2
 
+/obj/item/bombcore/emp
+	name = "EMP bomb core"
+	var/light_emp = 36
+	var/heavy_emp = 18
+	var/pulse_number = 1 //Since one EMP wont destroy anything other then consoles and IPCS, here is an option to have multiple pulses when dentonating. DO NOT USE THIS WITH REALLY LARGE AREAS
+	var/adminloged = FALSE //If it exploded once, don't do it again.
```
Comment (SteelSlayer) [inline code/game/machinery/syndicatebomb.dm:463]:
```suggestion
	var/adminlogged = FALSE //If it exploded once, don't do it again.
```

File: code/game/objects/items/devices/radio/beacon.dm [hunk 4]
```diff
@@ -76,15 +76,20 @@
 	name = "suspicious beacon"
 	desc = "A label on it reads: <i>Warning: Activating this device will send a high-ordinance explosive to your location</i>."
 	origin_tech = "bluespace=5;syndicate=5"
+	var/bomb = /obj/machinery/syndicatebomb
 
 /obj/item/radio/beacon/syndicate/bomb/attack_self(mob/user)
 	if(user)
 		to_chat(user, "<span class='notice'>Locked In</span>")
-		new /obj/machinery/syndicatebomb( user.loc )
+		new bomb( user.loc )
```
Comment (SteelSlayer) [inline code/game/objects/items/devices/radio/beacon.dm:84]:
```suggestion
		new bomb(user.loc)
```

**Review by SteelSlayer — COMMENTED** [review]
Spelling/nitpick stuff.

File: code/game/machinery/syndicatebomb.dm (hunk 3 above)
Comment (Qwertytoforty) [reply code/game/machinery/syndicatebomb.dm:463]: Need to change both variables, one moment

**Pull request merged**